clang/lib/Lex/ModuleMap.cpp

```diff
@@ -348,6 +348,10 @@ bool ModuleMap::resolveAsBuiltinHeader(
   if (!File)
     return false;
 
+  // Ensure the path to the module directory is absolute, otherwise
+  // builtin headers will fail to resolve when using relative resource
+  // directory paths without a -I.
+  llvm::sys::fs::make_absolute(Path);
```

Comment:
This should be using `FileManager::makeAbsolutePath` to ensure `-working-directory` is handled correctly, and so that it goes through VFS makeAbsolute instead of using the process working directory.